### ISSUE TITLE
ROC/AUC for multiclass not working?

### ISSUE THREAD

**Issue by DGaffney** [DistrictDataLabs/yellowbrick#288]
I read through the issue history and saw you've gone through and added in multiclass ROC/AUC scores, but after an hour of so of hacking along with the MNIST dataset I can't replicate. I'm seeing the following issue:

```
  File "/usr/local/lib/python2.7/site-packages/yellowbrick/classifier/rocauc.py", line 183, in score
    self.draw()
  File "/usr/local/lib/python2.7/site-packages/yellowbrick/classifier/rocauc.py", line 222, in draw
    color= self.colors[len(self.classes_)],
IndexError: list index out of range
```

The dataset is just a standard csv version from (https://pjreddie.com/media/files/mnist_train.csv), and the code I'm using to run it looks like:

```
from sklearn.ensemble import AdaBoostClassifier
from yellowbrick.classifier import ROCAUC
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=0.2)
visualizer = ROCAUC(AdaBoostClassifier(algorithm='SAMME.R', base_estimator=None,learning_rate=0.5, n_estimators=10, random_state=None), classes=list(set(y)))
visualizer.fit(X_train, y_train) 
visualizer.score(X_test, y_test)
visualizer.poof(outpath=diagnostic_image_path+"/roc_auc.png")
plt.clf()
```

Where `y` is a parsed set of labels from the csv and `x` is the parsed feature set. Does this issue make any sense?

**Comment by ndanielsen** [comment]
@DGaffney thanks for raising this issue.

**Comment by bbengfort** [comment]
@DGaffney yes, the issue makes sense and we've seen this issue pop up a couple of times but haven't fully gotten to the bottom of it yet. Try the following modification:

```python
visualizer = ROCAUC(AdaBoostClassifier(algorithm='SAMME.R', base_estimator=None,learning_rate=0.5, n_estimators=10, random_state=None))
```

E.g. removing the `classes` keyword argument from the visualizer. If that works, would you mind sending me a notebook with your work so I can dive deeper into the issue?

**Comment by DGaffney** [comment]
[110471_binary.pkl.zip](https://github.com/DistrictDataLabs/yellowbrick/files/1353545/110471_binary.pkl.zip)
The zipped pickle attached can work as follows:

```python
current_best_model, classes, X_train, X_test, y_train, y_test = pickle.loads(open("110471_binary.pkl").read())
visualizer = ROCAUC(current_best_model[0], classes=classes)
visualizer.fit(X_train, y_train) 
visualizer.score(X_test, y_test)
visualizer.poof(outpath="roc_auc.png")
```

For me, this leads to the same stack trace as posted above - either with classes present or removed. This is Python2.7, MPL 2.0.2, sklearn 0.18.1 - let me know if you need any other machine details:
```
Traceback (most recent call last):
  File "<stdin>", line 6, in <module>
  File "/usr/local/lib/python2.7/dist-packages/yellowbrick/classifier/rocauc.py", line 183, in score
    self.draw()
  File "/usr/local/lib/python2.7/dist-packages/yellowbrick/classifier/rocauc.py", line 222, in draw
    color= self.colors[len(self.classes_)],
IndexError: list index out of range
```

**Comment by bbengfort** [comment]
@gary-mayfield do you want to take a look at this?

**Comment by bbengfort** [comment]
@rebeccabilbro this might also relate to the bug described in #252

**Comment by bbengfort** [comment]
@rebeccabilbro I think this might have been fixed in #522 or #533; do you have time to use the mnist dataset and code that @DGaffney provided to see if this is resolved? If not, I can try to make time this weekend, let me know!

**Comment by DGaffney** [comment]
Beautiful to see forward movement in this ticket! Love what you are all putting together.

Devin

**Comment by rebeccabilbro** [comment]
@bbengfort - apologies for the delay! I can investigate tomorrow if that's ok, lmk

**Comment by bbengfort** [comment]
@rebeccabilbro thanks, that would be great! I'm working on the PyData Miami presentation this weekend; let me know if you need a rubber duck!

**Comment by rebeccabilbro** [comment]
@DGaffney @bbengfort - Sounds like the bug has been resolved! I just ran the pickle and above commands with no error, with the following plot as output:

![roc_auc](https://user-images.githubusercontent.com/8760385/50730147-0b699e00-1114-11e9-8368-81144557d9a8.png)
 
I did my best to replicate the environment reported above, but had to guess on a few of the library versions. Here's what I used:

Python==2.7.15
scipy==1.2.0
numpy==1.15.4
matplotlib==2.2.3
scikit-learn==0.20.2
yellowbrick==0.9

If this is satisfactory, I think it's safe to close this issue now.
